### ISSUE TITLE
Support for signing requests with JWT token

### ISSUE THREAD

**Issue by nbransby** [MunifTanjim/node-bitbucket#31]
According to these docs, if you want to call the API as an connect app you need to pass JWT in the query string or header param:

https://developer.atlassian.com/cloud/jira/service-desk/authentication-for-apps/

Also to create the JWT token in the first place according to these cdocs you need to include a query string hash: "A query string hash is a signed canonical request for the URI of the API you want to call.":

https://developer.atlassian.com/cloud/jira/service-desk/understanding-jwt/

Does node-bitbucket support this?

**Comment by MunifTanjim** [comment]
Those links are for JIRA.

Does [Bitbucket Cloud API](https://developer.atlassian.com/bitbucket/api/2/reference/) supports this?

**Comment by nbransby** [comment]
Yeah the docs are same https://developer.atlassian.com/cloud/bitbucket/authentication-for-apps/

**Comment by MunifTanjim** [comment]
Before this can be implemented, it'll need some architectural changes to the codebase. With the current architecture, I don't think it's possible.

**Comment by pavel-solodilov** [comment]
Raised a draft PR to address this: https://github.com/MunifTanjim/node-bitbucket/pull/117
